client-spark/spark2/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java

```diff
@@ -270,8 +270,8 @@ private void writeImpl(Iterator<Product2<K, V>> records) {
     long s = System.currentTimeMillis();
     checkAllBufferSpilled();
     checkSentRecordCount(recordCount);
+    checkBlockSendResult(new HashSet<>(blockIds));
```

Comment:
The variant of `blockIds` will be used for the `checkSentBlockIds` , so it's better to avoid modification in `checkBlockSendResult`